data_structures/linked_list.cpp

```diff
@@ -1,134 +1,280 @@
+/**
+ * @file linked_list.cpp
+ * @brief Implementation of singly linked list algorithm.
+ *
+ * @details
+ * The linked list is a form of container used for holding a sequence of
+ * values, which can be added, removed and displayed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list(by following
+ * the pointers) starting from the first link. Whichever link points to null
+ * is considered the last link and is pointed to the new value.
+ *
+ * Values can be removed by also iterating through the list. When the node
+ * containing the value is found, the node pointing to the current node is made
+ * to point to the node that the current node is pointing to, and then returning
+ * the current node to heap store.
+ */
 #include <iostream>
+#include <sstream>
 
-struct node {
-    int val;
-    node *next;
-};
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+/**
+ * @namespace linked_list
+ * @brief Functions for singly linked list algorithm
+ */
+namespace linked_list {
 
-node *start;
+/**
+ * This function changes the value_type of its argument
+ * from a string to an integer
+ * @param s argument whose value_type we want to change
+ * @return number value with value_type as integer
+ */
+int toInt(const std::string &s) {
```

Comment:
can be replaced by [`std::stoi`](https://en.cppreference.com/w/cpp/string/basic_string/stol) or an appropriate type-name alias